src/esp/assets/ResourceManager.cpp

```diff
@@ -568,38 +602,30 @@ std::string ResourceManager::getObjectConfig(int objectID) {
   return physicsObjectConfigList_[objectID];
 }
 
-void ResourceManager::shiftMeshDataToOrigin(GltfMeshData* meshDataGL) {
+
+Magnum::Vector3 ResourceManager::computeMeshBBCenter(GltfMeshData* meshDataGL){
   CollisionMeshData& meshData = meshDataGL->getCollisionMeshData();
-  float minX = 999999.9f;
-  float maxX = -999999.9f;
-  float minY = 999999.9f;
-  float maxY = -999999.9f;
-  float minZ = 999999.9f;
-  float maxZ = -999999.9f;
+  Magnum::Vector3 maxCorner(-999999.9), minCorner(999999.9);
   for (int vi = 0; vi < meshData.positions.size(); vi++) {
     Magnum::Vector3 pos = meshData.positions[vi];
-    if (pos.x() < minX) {
-      minX = pos.x();
-    }
-    if (pos.x() > maxX) {
-      maxX = pos.x();
-    }
-    if (pos.y() < minY) {
-      minY = pos.y();
-    }
-    if (pos.y() > maxY) {
-      maxY = pos.y();
-    }
-    if (pos.z() < minZ) {
-      minZ = pos.z();
-    }
-    if (pos.z() > maxZ) {
-      maxZ = pos.z();
-    }
+    maxCorner[0] = fmax(maxCorner.x(), pos.x());
+    maxCorner[1] = fmax(maxCorner.y(), pos.y());
+    maxCorner[2] = fmax(maxCorner.z(), pos.z());
+
+    minCorner[0] = fmin(minCorner.x(), pos.x());
+    minCorner[1] = fmin(minCorner.y(), pos.y());
+    minCorner[2] = fmin(minCorner.z(), pos.z());
   }
-  LOG(INFO) << "Shifting data origin";
-  Magnum::Matrix4 transform = Magnum::Matrix4::translation(Magnum::Vector3(
-      -(maxX + minX) / 2, -(maxY + minY) / 2, -(maxZ + minZ) / 2));
+
+  return (maxCorner+minCorner)/2.0;
```

Comment:
I feel like I already said this in another review of this exact code but can't find it right now -- this whole loop could be simplified to roughly the following, making use of [Math::minmax()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1Math.html#ac0524e916ab96970c6b00e99ecc6e912):

```cpp
#include <Corrade/Containers/ArrayViewStl.h>
#include <Magnum/Math/FunctionsBatch.h>

...

return Magnum::Range3D{Magnum::Math::minmax<Vector3>(meshData.positions)}.center();
```

In general, if you are ever looping over numbers to get their bounds or doing repetitive calculations over vector components like above, there's a high chance Magnum already has a well-tested function for that. Just search around, it'll save you *massive* amounts of time ;)